.drone.yml

```diff
@@ -1,32 +1,55 @@
+---
 kind: pipeline
-type: docker
+type: kubernetes
 name: default
 
-
 steps:
   - name: test
-    image: golang:1.13
+    image: golang:1.13-buster
     commands:
-      - go test ./cmd/... ./internal/...
-      - go vet ./cmd/... ./internal/...
+      - make test
+
   - name: lint
     image: cytopia/golint
     commands:
-      - golint -set_exit_status ./cmd/... ./internal/...
-  - name: build
-    image: golang:1.13
-    commands:
-      - go build -o build/drone-helm cmd/drone-helm/main.go
+      - golint -set_exit_status ./...
+
   - name: publish_linux_amd64
     image: plugins/docker
     settings:
-      auto_tag: true
+      registry: quay.io
+      repo: quay.io/mongodb/drone-helm
       username:
         from_secret: docker_username
       password:
         from_secret: docker_password
-      repo:
-        from_secret: plugin_repo
-      dockerfile: Dockerfile
+      tags:
+        - git-${DRONE_COMMIT_SHA:0:7}
+        - latest
     when:
-      event: [ tag, push ]
+      branch:
+        - master
+      event:
+        - push
+
+  - name: publish_linux_amd64_tag
+    image: plugins/docker
+    settings:
+      registry: quay.io
+      repo: quay.io/mongodb/drone-helm
+      username:
+        from_secret: docker_username
+      password:
+        from_secret: docker_password
+      tags:
+        - git-${DRONE_COMMIT_SHA:0:7}
+        - ${DRONE_TAG}
+        - latest
+    when:
+      event:
+        - tag
+---
+kind: signature
+hmac: 8329e42fe25499744eb7522048352e62e7bf776fc0db251d549c4ef2826c790d
```

Comment:
Any reason we're adding a signature here? We shouldn't be exposing any secrets to `pull_request` events.

Reply:
if this is not added then drone won't run when the pipeline is protected, or maybe I'm not understanding the usage for the signature

Reply:
i guess leaving it protected is a good idea since its a public repo. disregard.